apps/examples/testcase/ta_tc/st_things/stc/stc_st_things_main.c

```diff
@@ -0,0 +1,176 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <st_things/st_things.h>
+#include <iotbus/iotbus_gpio.h>
+#include "tc_common.h"
+
+#define TAG		"TESTS_THINGS"
+
+static const char *g_res_switch = "/capability/switch/main/0";
+static const char *g_res_dimming = "/capability/switchLevel/main/0";
+static const char *g_res_temp = "/capability/colorTemperature/main/0";
+
+extern void change_switch_value(void);
+extern bool handle_get_request_on_switch(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_get_request_on_dimming(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_set_request_on_dimming(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_get_request_on_ct(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_set_request_on_ct(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+
+#ifdef CONFIG_RESET_BUTTON
+static bool check_reset_button_pin_number(void)
+{
+	printf("Current pin number : %d\n", CONFIG_RESET_BUTTON_PIN_NUMBER);
+	return CONFIG_RESET_BUTTON_PIN_NUMBER >= 0 ? true : false;
+}
+
+static void gpio_callback_event(void *user_data)
+{
+	printf("gpio_callback_event!!\n");
+	printf("reset :: %d\n", st_things_reset());
+}
+#endif
+
+#ifdef CONFIG_LIGHT_SWITCH_BUTTON
+static void gpio_callback_event_trigger_light_switch(void *user_data)
+{
+	printf("gpio_callback_event to notify switch resource's observers manually!!\n");
+	change_switch_value();
+	printf("Notify to observer :: %d\n", st_things_notify_observers(g_res_switch));
+	set_rep_push_mesg();
+}
+#endif
+
+static void handle_reset_result(bool result)
+{
+	printf("[%s]Reset %s.\n", result ? "succeeded" : "failed", TAG);
```

Comment:
The order of the second and third parameters has changed.
`printf("[%s]Reset %s.\n", TAG, result ? "succeeded" : "failed");`

Reply:
Modified. Please check updated request.